testsuite/k8s-conformance-test-suite.sh

```diff
@@ -1,68 +1,187 @@
+    then
+        AZ_STORAGE_ACCOUNT="${line#*=}"
+
+    else
+        properties[$properties_count]=$line
+        ((properties_count++))
+    fi
+
+    if [[ $line == *"SUBSCRIPTION_ID"* ]]
+    then
+        AZ_SUBSCRIPTION_ID="${line#*=}"
+
+    elif [[ $line == *"CLIENT_ID"* ]]
+    then
+        AZ_CLIENT_ID="${line#*=}"
+
+    elif [[ $line == *"CLIENT_SECRET"* ]]
+    then
+        AZ_CLIENT_SECRET="${line#*=}"
+    
+    elif [[ $line == *"TENANT_ID"* ]]
+    then
+        AZ_TENANT_ID="${line#*=}"
+    fi
+
+done < /etc/config/azure-arc-conformance.properties
+# Parsing ends
+
+# Basic validation of required parameters
+error=0
+if [[ -z "${AZ_STORAGE_ACCOUNT_SAS}" ]]
+then
+    echo "Warning: Please set az-storage-account-sas in property file"
+fi
+
+if [[ -z "${AZ_STORAGE_ACCOUNT}" ]]
+then
+    echo "Warning: Please set az-storage-account in property file"
+fi
+
+if [[ -z "${AZ_SUBSCRIPTION_ID}" ]]
+then
+    echo "Error: Please set global.SUBSCRIPTION_ID in property file"
+    error=1
+fi
+
+if [[ -z "${AZ_CLIENT_ID}" ]]
+then
+    echo "Error: Please set global.CLIENT_ID in property file"
+    error=1
+fi
+
+if [[ -z "${AZ_CLIENT_SECRET}" ]]
+then
+    echo "Error: Please set global.CLIENT_SECRET in property file"
+    error=1
+fi
+
+if [[ -z "${AZ_TENANT_ID}" ]]
+then
+    echo "Error: Please set global.TENANT_ID in property file"
+    error=1
+fi
+
+if [[ error -eq 1 ]]
+then
+    exit
+fi
+# Parameter validation ends
 
 az login --service-principal --username $AZ_CLIENT_ID --password $AZ_CLIENT_SECRET --tenant $AZ_TENANT_ID
 az account set -s $AZ_SUBSCRIPTION_ID
 
+# Url will be changed test code
+echo "arc repository: $repository"
+git clone $repository
+cd /azure-arc-validation
+git checkout -b launcher_VMwareProposal origin/launcher_VMwareProposal
+cd /
+# test code ends
+
+plugins_files=(/azure-arc-validation/testsuite/conformance-test-plugins/*)
+
+command="sonobuoy run --wait"
+#check if yaml file is present for the user configuration
+add_helm=0
+found=0
+for enabled_plugin in "${enabled_plugins[@]}"
+do
+    for file in "${plugins_files[@]}"
+    do
+        if [[ $file == *$enabled_plugin* ]] 
+        then
+            found=1
+            if [[ $enabled_plugin == "azure-arc-platform" ]]
+            then
+                add_helm=1
+            fi
+            # prepare the command 
+            command="${command} --plugin $file" 
+            # get all the variables
+            for var in "${properties[@]}"
+            do
+                if [[ $var == $enabled_plugin* ]]
+                then
+                    command="${command} --plugin-env $var"
+                elif [[ $var == "global"* ]]
+                then
+                    plugin_var="${var/"global"/"$enabled_plugin"}"
+                    command="${command} --plugin-env $plugin_var"
+                fi
+            done            
+            break
+        fi
+    done
+    if [[ $found -eq 0 ]]
```

Comment:
Shouldn't we reset variable 'found' to value '0' again after competion of the loop?

Reply:
ack